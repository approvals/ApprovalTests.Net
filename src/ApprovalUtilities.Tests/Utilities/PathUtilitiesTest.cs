using System.Linq;
using ApprovalTests;
using ApprovalUtilities.Utilities;
using Xunit;

namespace ApprovalUtilities.Tests.Utilities
{
    public class PathUtilitiesTest
    {
        [Fact]
        public void ScrubPathTest()
        {
            var dir = PathUtilities.GetDirectoryForCaller();
            var file = dir + "PathUtilitiesTest.cs";
            AssertEqualIgnoreCase(@"...\PathUtilitiesTest.cs", file.ScrubPath(dir));
        }

        [Fact]
        public void TestFindsFile()
        {
            var found = PathUtilities.LocateFileFromEnvironmentPath("ipconfig.exe").FirstOrDefault();
            AssertEqualIgnoreCase(@"C:\Windows\System32\ipconfig.exe", found);
        }

        private void AssertEqualIgnoreCase(string expected, string actual)
        {
            Assert.Equal(expected.ToLowerInvariant(), actual.ToLowerInvariant());
        }

        [Fact]
        public void TestFindsMultipleFiles()
        {
            Approvals.VerifyAll(PathUtilities.LocateFileFromEnvironmentPath("notepad.exe").Select(f => f.ToLowerInvariant()), "Found");
        }

        [Fact]
        public void TestDoesNotFindFile()
        {
            var noneExistingFile = "ThisFileShouldNotExist.exe";
            var results = PathUtilities.LocateFileFromEnvironmentPath(noneExistingFile);
            Assert.Empty(results);
        }
    }
}