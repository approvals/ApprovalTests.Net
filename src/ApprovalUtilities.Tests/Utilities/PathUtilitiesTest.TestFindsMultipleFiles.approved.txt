Found[0] = c:\windows\system32\notepad.exe
Found[1] = c:\windows\notepad.exe
